requirement-generator/requirement-satisfiable-course-generator.ts

```diff
@@ -17,25 +24,50 @@ function courseCanSatisfyCollegeOrMajorRequirement(
 
 const produceSatisfiableCourseIdForCollegeOrMajorRequirement = (
   collegeRequirement: CollegeOrMajorRequirement
-): readonly string[] => {
-  const set = new Set<string>();
+): EligibleCourses => {
+  const allMap: { [semester: string]: { [subject: string]: string[] } } = {};
```

Comment:
Can this variable be named a little bit better? I wasn't sure what this is or why it is returned at first glance. `allElidgibleCoursesSemestersMap` would be much better